compiler/src/main/kotlin/dev/zacsweers/metro/compiler/fir/generators/ContributedInterfaceSupertypeGenerator.kt

```diff
@@ -113,14 +114,19 @@ internal class ContributedInterfaceSupertypeGenerator(session: FirSession) :
             .mapNotNull { it.resolvedScopeClassId(typeResolver) }
             .distinct()
             .forEach { scopeClassId ->
-              val metroContribution =
-                originClass.classId.createNestedClassId(Symbols.Names.metroContribution)
+              val metroContribution = originClass.nestedContributionClassId(scopeClassId)
               getOrPut(scopeClassId, ::mutableSetOf).add(metroContribution)
             }
         }
       }
     }
 
+  private fun FirRegularClassSymbol.nestedContributionClassId(scopeClassId: ClassId): ClassId {
+    return classId.createNestedClassId(
+      Symbols.Names.metroContribution.plus(scopeClassId.shortClassName.identifier)
```

Comment:
I worry this case, however unlikely, would be an issue. What do you think of using `NameAllocator` to allocate names deterministically in this scenario so we avoid that? If we combine that with using a holder annotation to hold the real scope, we should be able to still retrieve the "real" scope key without worrying about what the name of the holding contribution class is.

Reply:
> I worry this case, however unlikely, would be an issue. What do you think of using NameAllocator to allocate names deterministically in this scenario so we avoid that? If we combine that with using a holder annotation to hold the real scope, we should be able to still retrieve the "real" scope key without worrying about what the name of the holding contribution class is.

Not sure if some of the text got cut off but are you thinking of the case where there are multiple scopes in different packages but with the same name? Not familiar with `NameAllocator` so I need to take a look at that but switching to a different naming pattern sounds reasonable to me when coupled with an annotation to store the scope info.

Reply:
Yup exactly, same simple name but different packages